sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch_app_sdk/TaskResultUploader.java

```diff
@@ -0,0 +1,123 @@
+import org.sagebionetworks.bridge.android.manager.UploadManager;
+import org.sagebionetworks.bridge.android.manager.dao.AccountDAO;
+import org.sagebionetworks.bridge.android.manager.upload.ArchiveUtil;
+import org.sagebionetworks.bridge.android.manager.upload.SchemaKey;
+import org.sagebionetworks.bridge.data.Archive;
+import org.sagebionetworks.bridge.data.ArchiveFile;
+import org.sagebionetworks.bridge.data.JsonArchiveFile;
+import org.sagebionetworks.bridge.rest.model.Activity;
+import org.sagebionetworks.bridge.rest.model.ScheduledActivity;
+import org.sagebionetworks.bridge.rest.model.TaskReference;
+import org.sagebionetworks.research.domain.result.interfaces.TaskResult;
+import org.sagebionetworks.research.presentation.perform_task.TaskResultProcessingManager.TaskResultProcessor;
+import org.sagebionetworks.research.sageresearch_app_sdk.archive.AbstractResultArchiveFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Locale;
+
+public class TaskResultUploader implements TaskResultProcessor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskResultUploader.class);
+    private final BridgeConfig bridgeConfig;
+    private AbstractResultArchiveFactory abstractResultArchiveFactory;
+    private AccountDAO accountDAO;
+    private UploadManager uploadManager;
+
+    @Inject
+    public TaskResultUploader(final BridgeConfig bridgeConfig,
+                              final AbstractResultArchiveFactory abstractResultArchiveFactory,
+                              final UploadManager uploadManager,
+                              final AccountDAO accountDAO) {
+        this.bridgeConfig = bridgeConfig;
+        this.abstractResultArchiveFactory = abstractResultArchiveFactory;
+        this.uploadManager = uploadManager;
+        this.accountDAO = accountDAO;
+    }
+
+    @Override
+    public Completable processTaskResult(final TaskResult taskResult) {
+        LOGGER.info("Processing task result: {}", taskResult);
+
+        // TODO: wrap operations in completable
+        SchemaKey sk = bridgeConfig.getTaskToSchemaMap().get(taskResult.getIdentifier());
+
+        if (sk == null) {
+            LOGGER.info("No schema key found for task with identifier: {}, skipping upload",
+                    taskResult.getIdentifier());
+            // TODO: use taskId/1 as default for schema/revision?
+            return Completable.complete();
+        }
+
+        Archive.Builder builder = Archive.Builder.forActivity(sk.getId(), sk.getRevision());
+        String appVersionString = String.format(Locale.ENGLISH, "version %s, build %d",
+                bridgeConfig.getAppVersionName(),
+                bridgeConfig.getAppVersion());
+
+        builder.withAppVersionName(appVersionString)
+                .withPhoneInfo(bridgeConfig.getDeviceName());
+
+        //TODO: load scheduled activity
+        ScheduledActivity sa = new ScheduledActivity();

```

Comment:
I think you could just do a query on guid, then use RestUtils.gson to convert that to ScheduledActivity

Reply:
Oops, I didn't finish this part yet. Good catch. What I need to do is call the TaskRunDao, which stores taskRunUUID -> ScheduledActivity guid and do as you mentioned